### PR TITLE
639 qa filter

### PR DESCRIPTION
**Pull request by marksmall**
I've added a new filter to the sidebar to deal with Quality Assurance as requested by Alasdair. This filter does nothing right now but set a new option on the `activeFilters` object called `qualityAssurance`.

### CODE REVIEW

File: src/components/FilterSidebar/components/QAFilter.test.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { FilterData } from '@/context/FilterContext/types'; // adjust if needed
+import { render, screen, userEvent } from '@/utils/renderers';
+
+import { QAFilter } from './QAFilter'; // adjust the import based on your file structure
```
Comment (james-hinton) [inline src/components/FilterSidebar/components/QAFilter.test.tsx:4]: Just some small comments that could be confusing

File: src/components/FilterSidebar/components/QAFilter.test.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+// Mock data for the FilterData type
+const filterData: FilterData = {
+  name: 'Test Filter',
+  type: 'combobox',
+  options: [
+    { value: 'pass', label: 'Pass' },
+    { value: 'partial', label: 'Partial' },
+    { value: 'fail', label: 'Fail' },
```
Comment (james-hinton) [inline src/components/FilterSidebar/components/QAFilter.test.tsx:13]: Should also test with the 'none' option as well

Comment (james-hinton) [reply src/components/FilterSidebar/components/QAFilter.test.tsx:13]: I can see it as an option [here](https://github.com/EO-DataHub/eodhp-resource-catalogue-ui/pull/17/files#diff-7c2a224ea560b6b43a8d0b9a34fc9aa304c079f669d633055cd2e3333df64295R42)

File: src/components/FilterSidebar/components/QAFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export const QAFilter = ({ filterData, onChange }: QAFilterProps) => (
+  <div>
+    <label htmlFor="options">{filterData.name}</label>
+
+    <select id="options" onChange={onChange}>
+      <option value="none">Please select an option</option>
+
+      {filterData.options.map((option) => (
+        <option key={option.value} value={option.value}>
```
Comment (james-hinton) [inline src/components/FilterSidebar/components/QAFilter.tsx:18]: We should also pass what value is selected, especially if we are 'storing' the filter state via URL query params in the future.

File: src/components/FilterSidebar/components/QAFilter.test.tsx (hunk 1 above)
Comment (marksmall) [reply src/components/FilterSidebar/components/QAFilter.test.tsx:4]: I'll remove these.

File: src/components/FilterSidebar/components/QAFilter.test.tsx (hunk 2 above)
Comment (marksmall) [reply src/components/FilterSidebar/components/QAFilter.test.tsx:13]: That option should never be used, that is just a value used for `Please select an option`. There are only 2 possibilities:

* Auto-select a real value
* Have a fake option with the label informing the user to select one and check it has been sent when submitting.

I don't want to be adding error handling at every option, especially if it is data-driven as you've built the sidebar i.e. mock `Product Type`, `Licence` etc. If it is dynamic, how do we validate the options selected?

I will change it so that the first option is auto-selected, that way the filter is never invalid, which is probably better than having a fake option, but some prefer that is all.

File: src/components/FilterSidebar/components/QAFilter.tsx (hunk 3 above)
Comment (marksmall) [reply src/components/FilterSidebar/components/QAFilter.tsx:18]: Can you clarify what you mean by`pass what value is selected`? Pass it where?

Right now filter state is in context, it should either be in context, or in the URL, not both (in my opinion).

Comment (james-hinton) [reply src/components/FilterSidebar/components/QAFilter.tsx:18]: For when the user flows passes through the Keycloak auth flow, we will lose the context and it will have to be repopulated.

In the events that happens, none of the `<options>` will be selected.

File: src/components/FilterSidebar/components/QAFilter.test.tsx (hunk 2 above)
Comment (james-hinton) [reply src/components/FilterSidebar/components/QAFilter.test.tsx:13]: > If it is dynamic, how do we validate the options selected?

Yeah it's a big grey area for me as well, I split it into the input types, so at least we can do some very basic validation based upon the type. 

However, it seems like this validation logic shouldn't lie in the frontend anyway, when whomever gets around to building this microservice that can return filters based upon the data, then perhaps that can also send some type of validation checks.

**Review by james-hinton — APPROVED** [review]

File: src/components/FilterSidebar/components/QAFilter.tsx (hunk 3 above)
Comment (marksmall) [reply src/components/FilterSidebar/components/QAFilter.tsx]: I'm not going to do this right now as that should be done altogether for all filters. 

There is a ticket to reset `filters` after auth, that is the appropriate ticket for that to come under. It will mean removing the current `FilterContext` and moving any state there to the URL as a `queryString`. The client is wrapped in a `react-router-dom` Provider, so we can then use `UseSearchParams` hook wherever we currently use `useFilters`.
